New recipe: Add icons to external links (#2970)
withastro/docs: src/content/docs/en/recipes/external-links.mdx

---

> title: Add icons to external links

Reviewer: ```suggestion
title: Add icons to external links
```

> description: Learn how to add icons to external links in your Markdown files

Reviewer: ```suggestion
description: Learn how to install a rehype plugin to add icons to external links in your Markdown files
```

i18nReady: false
type: recipe
---

> Using a Rehype plugin, you can add icons to external links in your Markdown files, as well as open those links in a new tab.

Reviewer: ```suggestion
Using a rehype plugin, you can identify links in your markdown files that point to external sites and modify them. This includes adding icons to the end of each link.
```

## Prerequisites
- A project using Markdown for content pages.

## Recipe

1. Install the `rehype-external-links` plugin.

    ```shell
    npm install rehype-external-links
    ```

> 2. Add the plugin to your `astro.config.mjs` file. Pass the plugin options by passing an array with the plugin as the first item and the options as the second item. Set the content type to `text`, or `raw` to use HTML as the content.

Reviewer: ```suggestion
2. Import the plugin into your `astro.config.mjs` file. Pass it to the `rehypePlugins` array, along with an options object. Add a content property and set its `type` to `text` if you want to add plain text to the end of the link, or `raw` to add HTML to the end of the link.
```

    ```ts
    // ...
    import rehypeExternalLinks from 'rehype-external-links';

    export default defineConfig({
      // ...
      markdown: {
        rehypePlugins: [
          [
            rehypeExternalLinks,
            {

>               target: '_blank',
>               rel: ['noopener', 'noreferrer'],
>               content: { type: 'text', value: ' 🔗' }

Reviewer: ```suggestion
              content: { type: 'text', value: ' 🔗' }
```

For the purposes of this example, we don't need to set these attributes to add an icon.

Additionally, the package warns us to not set `target`, and setting `rel` to these values is only appropriate if `target` is `_blank`.
